pkg/cmd/admin/validate/master.go

```diff
@@ -0,0 +1,172 @@
+		Run: func(c *cobra.Command, args []string) {
+			if err := options.Complete(args, f); err != nil {
+				cmdutil.CheckErr(cmdutil.UsageError(c, err.Error()))
+			}
+
+			err, failure := options.Run()
+			if err != nil {
+				cmdutil.CheckErr(cmdutil.UsageError(c, err.Error()))
+			}
+			if failure {
+				options.Out.Write([]byte(fmt.Sprintf("FAILURE: Validation failed for file: %s\n", options.MasterConfigFile)))
+				os.Exit(1)
+			}
+
+			options.Out.Write([]byte(fmt.Sprintf("SUCCESS: Validation succeded for file: %s\n", options.MasterConfigFile)))
+		},
+	}
+
+	cmdutil.AddPrinterFlags(cmd)
+
+	return cmd
+}
+
+func (o *ValidateMasterConfigOptions) Complete(args []string, f *clientcmd.Factory) error {
+	switch len(args) {
+	case 0:
+		return errors.New("the path to the file being validated is required")
+	case 1:
+		o.MasterConfigFile = args[0]
+		return nil
+	default:
+		return errors.New("at most one file can be identified for validation")
+	}
+}
+
+func (o *ValidateMasterConfigOptions) Run() (error, bool) {
+	masterConfig, err := configapilatest.ReadAndResolveMasterConfig(o.MasterConfigFile)
+	if err != nil {
+		return err, false
+	}
+
+	results := validation.ValidateMasterConfig(masterConfig)
+	writer := tabwriter.NewWriter(o.Out, minColumnWidth, tabWidth, padding, padchar, flags)
+	err = prettyPrintValidationResults(results, writer)
+	if err != nil {
+		return fmt.Errorf("could not print results: %v", err), len(results.Errors) > 0
+	}
+	writer.Flush()
+	return nil, len(results.Errors) > 0
+}
+
+const (
+	minColumnWidth          = 4
+	tabWidth                = 4
+	padding                 = 2
+	padchar                 = byte(' ')
+	flags                   = 0
+	validationErrorHeadings = "ERROR\tFIELD\tVALUE\tDETAILS\n"
+)
+
+// prettyPrintValidationResults prints the contnents of the ValidationResults into the buffer of a tabwriter.Writer.
+// The writer must be Flush()ed after calling this to write the buffered data.
+func prettyPrintValidationResults(results validation.ValidationResults, writer *tabwriter.Writer) error {
+	if len(results.Errors) > 0 {
+		writer.Write([]byte("VALIDATION ERRORS:\t\t\t\n"))
+		err := prettyPrintValidationErrorList(results.Errors, writer)
+		if err != nil {
+			return err
+		}
+	}
+	if len(results.Warnings) > 0 {
+		writer.Write([]byte("VALIDATION WARNINGS:\t\t\t\n"))
+		err := prettyPrintValidationErrorList(results.Errors, writer)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// prettyPrintValidationErrorList prints the contnents of the ValidationErrorList into the buffer of a tabwriter.Writer.
+// The writer must be Flush()ed after calling this to write the buffered data.
+func prettyPrintValidationErrorList(validationErrors fielderrors.ValidationErrorList, writer *tabwriter.Writer) error {
+	if len(validationErrors) > 0 {
+		writer.Write([]byte(validationErrorHeadings))
+		for _, err := range validationErrors {
+			validationError := err.(*fielderrors.ValidationError)
```

Comment:
this will panic if the type isn't what we expect... do we want to do a conditional cast and output err.Error() if it isn't something we recognize?